thijsvanloef/palworld-server-docker
# add unit test

## Context 

* resolves https://github.com/thijsvanloef/palworld-server-docker/issues/343#issue-2127691853
* Add unit test in `uni-test.yml`
* Test if the server has successfully started through RCON.
* Test if port 8211, 27015 and 25575 are listening with `netcat` command.
* Test the backup command functions and check if the `./palworld/backups/` path contains a backup folder.

~~Due to the limitation that macOS environments in GitHub Actions do not support Docker for `ARM64`, it cannot be included.~~ 
~~Refer to: https://github.com/orgs/community/discussions/25777#discussioncomment-3249229~~

Not sure how to check If PalWorldSettings.ini is valid.

## Choices

* Improving Efficiency Through CI

## Test instructions

1. Push to the main branch and check each unit test step to ensure it meets the expected criteria.


## Checklist before requesting a review
